x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java

```diff
@@ -104,14 +101,7 @@ public boolean hasMetric(String name) {
 
     @Override
     public double metric(String name, long owningBucketOrd) {
-        assert size == 1;
-        /*
-         * Since size is always 1 we know that the index into the values
-         * array is same same as the bucket ordinal. Also, this will always
-         * be called after we've collected a bucket, so it won't just fetch
-         * garbage.
-         */
-        return metrics.metric(name, owningBucketOrd);
+        return metrics.metric(name, owningBucketOrd * size);
```

Comment:
I don't get why we here compute index as `owningBucketOrd * 2`.

Reply:
Also, if we always take the latest...since `owningBucketOrd * size` should be always the latest element in the array, I guess the actual value will depend on whether data is sorted in ascending or descending order.

Reply:
It's the *first* element in the array.